source/configuration/modules/omfile.rst

```diff
@@ -1,434 +1,441 @@
+================  ================
+fileGroupNum:     "gid (number)"
+Default:          *system default*
+Available since:  7.5.8, 8.1.4
+================  ================
 
-   *Default: on*
+**TODO why available since 2 versions? backport? bugfixes?**
 
-   This option modifies behaviour of file creation. If different owners
-   or groups are specified for new files or directories and rsyslogd
-   fails to set these new owners or groups, it will log an error and NOT
-   write to the file in question if that option is set to "on". If it is
-   set to "off", the error will be ignored and processing continues.
-   Keep in mind, that the files in this case may be (in)accessible by
-   people who should not have permission. The default is "on".
+Set the group for **newly created files**.
+This setting does not affect the group of already existing files.
 
-.. function::  createDirs [switch]
+The parameter is a numerical ID, which is used regardless
+of whether the group actually exists. This can be useful if the group
+mapping is not available to rsyslog during startup.
 
-   *Default: on*
+===============  ===============
+fileCreateMode:  "octalnumber"
+Default:         same as module
+Legacy:          $FileCreateMode
+===============  ===============
 
-   create directories on an as-needed basis
+Allows to specify the file creation mask with which rsyslogd creates new files.
+If not specified, the module configuration is used.
 
-.. function::  sync [switch]
+==============  ==============
+dirCreateMode:  "octalnumber"
+Default:        same as module
+Legacy:         $DirCreateMode
+==============  ==============
 
-   *Default: off*
+The same as fileCreateMode, but for automatically generated directories.
 
-   enables file syncing capability of omfile.
+===================  ===================
+failOnChOwnFailure:  "on|off"
+Default:             "on"
+Legacy:              $FailOnCHOwnFailure
+===================  ===================
 
-   When enabled, rsyslog does a sync to the data file as well as the
-   directory it resides after processing each batch. There currently
-   is no way to sync only after each n-th batch.
+This option modifies behaviour of file creation.
 
-   Enabling sync causes a severe performance hit. Actually,
-   it slows omfile so much down, that the probability of loosing messages
-   **increases**. In short,
-   you should enable syncing only if you know exactly what you do, and
-   fully understand how the rest of the engine works, and have tuned
-   the rest of the engine to lossless operations.
+If different owners or groups are specified for new files or directories, and rsyslogd fails to set them:
 
-.. function::  sig.provider [providerName]
+* If set to *on*, it will log an error and NOT write to the file.
+* If it is set to *off*, the error will be ignored and processing continues.
+  Keep in mind that in the event of an error, files may be (in)accessible due to permissions.
```

Comment:
by inaccessible, this means that rsyslog may not be able to write to the file.

there should probably be a section of failure modes (disk full, permissions, etc). we talk about permissions a bit in the dynafile section, but it really applies to everything.